### DISCUSSION TITLE
Queue view doesn't necessarily correspond with the mpd's queue

### DISCUSSION THREAD

**Discussion by mgonzalezm** [jcorporation/myMPD#966] [Ideas]
In the queue view we can sort the order of the songs according to the different columns at the top, but this is not reflected in the order of the mpd queue. If the 'Pos' column is not shown (  this column can be disabled in the gear icon ) this can lead to confusion.

```
$ mpc --host=/run/mpd.socket playlist
The Black Keys - Gold on the Ceiling
B.B. King - The Other Night Blues
Neil Young - Don't Let It Bring You Down
The Rolling Stones - Oh No, Not You Again
Manu Chao - Día luna... Día pena
Era - The Mass
Fleetwood Mac - Illume (9-11)
Pearl Jam - Light Years
The Kinks - You Really Got Me (mono LP version with reverb)

```

![screenshot](https://user-images.githubusercontent.com/11181196/216786338-0ca8851b-23e6-424a-a079-0d00356db14b.png)

I think that or this reordering should not be permitted or that the mpd's queue be updated accordingly.

what do you think?

PS. The little triangle besides the column name is also a little confusing because it is closer to the next column name,  in my screenshot at first glance I thought the order was by  album instead of by artist, but maybe that is just me

**Comment by jcorporation** [comment]
This is a new feature of MPD 0.24 no special myMPD implementation:
https://github.com/MusicPlayerDaemon/MPD/blob/master/NEWS
`- "playlistfind"/"playlistsearch" have "sort" and "window" parameters`

**Reply by mgonzalezm** [reply]
right, as i understand it, the purpose of those options is for presentation only, but then there is no way to known exactly what the next songs to be played will be.

this took me by surprise because in a mobile screen I was normally disabling all columns except for the artist and title to save screen space, and besides this reordering is working different to when we reorder by drag and drop where the mpd queue IS changed.

what about not allowing to hide the position column? or even better a button to apply the new ordering to the queue?

**Reply by jcorporation** [reply]
Reordering could be disabled if queue is not sorted by position.

**Reply by mgonzalezm** [reply]
well yes that will fix part of the odd things with the queue view, will you force the position to be always shown ?

if not, then additionally would I also like the sorting to be disabled if it is not impacting the mpd's queue, because if by mistake you click on a column there is no way to recover the order of the queue apart from un-hiding the pos column and sort by it. Not even navigating away from the queue and returning works.

**Reply by jcorporation** [reply]
An option to disable the queue sorting could be implemented.
